AdamSyed/movie-finder
# Adam

What was done?
1. The largest story I took on was building the navbar. I leveraged Bulma's CSS framework (an open source framework that provides web components to use - similar to Bootstrap). The navbar was placed on several pages, but only links to the rating, user profile, and login pages. Other pages will be accessible from these pages. As part of this, I edited the HTML and CSS to make the pages more standardized and appealing.
2. It was pointed out that we did not perform sufficient unit testing on the create-account page in a previous sprint. To handle this, I performed unit testing on the page and ended up fixing two bugs on the page. Firstly, I changed the code so that users needed to fill all the fields before submitting. Secondly, I checked if the email inserted already corresponded to an account. If it did, I informed the user that an account already exists with this email.
3. I pair-programmed with Jishaan on the edit-preferences page. I helped conceptualize how he should handle the dynamic iteration to read in the results when a user clicks "Update". When doing this we realized that the user-preferences endpoint was only passing the movie name, when it should also be passing the movieID. I changed the endpoint to reflect this.
4. Finally, I updated the user-profile page to also check if all fields have been filled. If not, it informs the user to go back and fill all fields.

Areas that need review?
1. Web design. What do you think of the colours and design choices I made? Do you think I need to make any changes? I've tested every page with different userIDs. The pages seem to have appropriate spacing with the information from these UserIDs, but perhaps you can test this with creating your own user and inserting preferences to see how the frontend handles that.
2. I've used the open-source framework Bulma to create the navbar. Do you think I've provided enough credit to them? Or should I add a comment block of code above the index.html page to give further credit to them?

How was this code tested?  What is the proof that you are providing working code?
1. The nabar was tested thoroughly during development. On each page it was inserted, I applied unit tests to ensure each button on the navbar linked to the correct page and carried the relevant data (i.e. IDs). For example, when the navbar was placed onto the rating.html page, I tested to make sure the buttons correctly redirected to the rating.html page, user-profile.html page, and index.html page. The proof that this is working is that all redirects change the user's window to the correct page. Integration tests were also applied on each page to ensure that the navbar and new design did not interfere with the page's other functionality (i.e. cutting off any information or the information not being formatted correctly). For example, on the rating.html page I had to make sure that the buttons didn't extend past the background.
2. On the create-account page I performed a lot of unit tests. Not only did I fix the problem that was outlined, the fact that a user could be created with no information submitted, I also identified a new unit test. The new unit test was to see if a user could insert the same email as an account already created. After testing I realized that this could happen. I made changes to the backend and frontend to perform this check before creating an account.
3. On the backend change to the edit-preferences endpoint I performed unit testing using Postman to see if all requests would return movieName AND movieID. I couldn't perform integration testing with the frontend until Jishaan finishes building the frontend. Once he's done, we will both perform the integration testing.
4. I performed unit testing to ensure that users have to fill out all fields on the page in order for the user-profile page to update the user's profile. One weird unit test that I performed was to see if the user only entered a password and no other information if it would accept the request. It did not accept the request and prompted the user to fill in the other details.

How can you test the code?
1. I recommend creating a user and clicking all the buttons thoroughly. This way you will test it with new IDs and user data to ensure that the design still displays all user information correctly.
2. You can try to create an account with an email already created. It shouldn't let you do so.
3. Once Jishaan's done his section you can test his page. If it displays and submits correctly, then this endpoint worked.
4. You can try to edit a user's profile with not all of the fields filled. If it does not let you update the profile, then this worked.
